### PR TITLE
Add new Wallets! station trait

### PR DESCRIPTION
**Pull request by coiax**
> Add a new simple station trait, that replaces everyone's ID with a
> wallet containing said ID, as well as their starting money, and if
> they're lucky, a little bonus surprise.
> 
> - In addition, adds a `force` variable to station traits, which will
>   always enable them, rather than having to edit weight and probability
>   of rolling a trait of that sign (postive, negative).

---

## Why It's Good For The Game

1. It's a small station trait to surprise people occasionally, but doesn't drastically change the round.
2. It provides camoflague for antagonists who want to use wallets to store multiple IDs.

## Changelog
:cl: coiax, Timberpoes
expansion: Adds the new Wallets! station trait. Makes everyone start with wallets with
their ID card inside, along with their starting budget. Might also include a little
surprise.
expansion: Wallets that spawn inside dorm wardrobes also have little surprises in them.
expansion: Wallets can now hold pills.
/:cl:

---

- [x] Work out why people are spawning with no ID card at all

### CODE REVIEW

File: code/datums/station_traits/positive_traits.dm
Comment (coiax) [inline code/datums/station_traits/positive_traits.dm:228]: Note to other maintainers: Do not merge this without having this changed to FALSE.

**Review by Timberpoes — COMMENTED** [review]
Look at using wallet.InsertID(id_card) to insert the ID card into the wallet. It returns TRUE if the ID card was successfully inserted into the wallet, so you can handle conditional logic and debug it from there.

For order of operations, I'd unequip the ID card, equip the wallet and then insert the ID card into the wallet if the equip was successful. `equip_to_slot_if_possible()` returns TRUE or FALSE depending on success, so you can always use that to check success and `stack_trace()` when something unexpected occurs (useful for failing unit tests on error states too).

The above logic order means that you don't risk the wallet getting deleted and losing the ID card to go with it. By convention, qdel(thing) should also qdel its contents. You'll always retain control of the ID card until the final point and can always re-equip it if everything else fails. Same with the holochips, can always be added to the wallet after the ID card has been inserted.

File: code/datums/station_traits/positive_traits.dm
Comment (Timberpoes) [inline code/datums/station_traits/positive_traits.dm:249]:
```suggestion
	id_card.forceMove(wallet)
```
Since I know what the problem is now, in retrospect we don't need InsertID anymore (we could use it via INVOKE_ASYNC but all the logic is handled when `forceMove`ing as well.)

**Review by Timberpoes — CHANGES_REQUESTED** [review]
image

Add this in `/obj/item/storage/wallet/update_overlays()`

The issue with the wallet icon not updating: We're creating a new ID card in the equip procs, applying the trim, updating overlays - But this only queues the overlays for building in SSoverlays and doesn't ACTUALLY build them. Insert the ID card into the wallet. Wallet inherits ID card's overlays (which don't exist as they haven't been built yet) and thus we get a blank looking ID in the wallet.

If we sneak in a cheeky COMPILE_OVERLAYS, we can always make sure the id_card's overlays are built as part of the wallet's overlay update process.

**Comment by Bawhoppen** [comment]
As a station quirk, it's kind of a weird logic-wise, but it's so minor that no big deal on that

File: code/datums/station_traits/positive_traits.dm
Comment (Mothblocks) [reply code/datums/station_traits/positive_traits.dm]: well then remove it :P

**Comment by coiax** [comment]
> 
> 
> As a station quirk, it's kind of a weird logic-wise, but it's so minor that no big deal on that

I don't understand, could you rephrase?

File: code/datums/station_traits/positive_traits.dm
Comment (Rohesie) [inline code/datums/station_traits/positive_traits.dm:258]:
```suggestion
	for(var/obj/item/item in wallet)
```
The `contents` list may generally contain any kind of movable atoms.
It should probably use a different kind of list to store contents, but that's a bigger refactor.
In the meantime I think type checking should not be skipped here.
As for `in wallet` vs `in wallet.contents`, that's just stylistic, do whatever you prefer. I tend not to add the contents part because it's already implied.

**Review by Rohesie — APPROVED** [review]
Pending Timber's approval.

**Pull request merged**
